extensions-core/kafka-extraction-namespace/src/main/java/io/druid/query/lookup/KafkaLookupExtractorFactory.java

```diff
@@ -375,29 +361,85 @@ public LookupExtractor get()
     };
   }
 
-  public long getCompletedEventCount()
-  {
-    return doubleEventCount.get() >> 1;
+  private boolean advanceState(State expectedCurrentState, State newState) {
+    boolean success = currentState.compareAndSet(expectedCurrentState, newState);
+
+    if(success)
+    {
+      synchronized (currentState)
+      {
+        currentState.notifyAll();
+      }
+    }
+
+    return success;
   }
 
-  // Used in tests
-  NamespaceExtractionCacheManager getCacheManager()
-  {
-    return cacheManager;
+  private boolean awaitState(State expectedState, long timeout) throws InterruptedException {
+    final long waitStart = System.currentTimeMillis();
+
+    while(!currentState.get().equals(expectedState) && System.currentTimeMillis() - waitStart < timeout)
+    {
+      synchronized (currentState)
+      {
+        currentState.wait(timeout);
+      }
+    }
+
+    return currentState.get().equals(expectedState);
   }
 
-  AtomicReference<Map<String, String>> getMapRef()
-  {
-    return mapRef;
+  private boolean shouldStop() {
+    State state = currentState.get();
+
+    return state.equals(State.STOPPING) || state.equals(State.TERMINATED) || state.equals(State.FAILED);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+
+    KafkaLookupExtractorFactory that = (KafkaLookupExtractorFactory) o;
+
+    if (connectTimeout != that.connectTimeout) return false;
+    if (injective != that.injective) return false;
+    if (!kafkaTopic.equals(that.kafkaTopic)) return false;
+    return kafkaProperties.equals(that.kafkaProperties);
+
+  }
+
+  @Override
+  public int hashCode() {
+    int result = kafkaTopic.hashCode();
+    result = 31 * result + kafkaProperties.hashCode();
+    result = 31 * result + (int) (connectTimeout ^ (connectTimeout >>> 32));
+    result = 31 * result + (injective ? 1 : 0);
+    return result;
   }
 
-  AtomicLong getDoubleEventCount()
+  public long getCompletedEventCount()
   {
-    return doubleEventCount;
+    return doubleEventCount.get() >> 1;
+  }
+
+  public State getState() {
+    return currentState.get();
   }
 
-  ListenableFuture<?> getFuture()
+  interface ConsumerConnectorFactory
```

Comment:
The factory was added to avoid extending the class in tests and also to ensure that the method is not called when it shouldn't